### PR TITLE
chore(amplify): missing integ test for AWS Amplify

### PR DESCRIPTION
**Pull request by SankyRed**
Adding an integ test for this custom resource with our new rule that each custom resource needs at least one integ test. Adding this test for the custom resource to test out AWS Amplify app.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
   }
 }
 
 const app = new App();
-new TestStack(app, 'cdk-amplify-app-asset-deployment');
-app.synth();
+const stack = new TestStack(app, 'cdk-amplify-app-asset-deployment');
+new IntegTest(app, 'cdk-amplify-app-integ-test', {
+  testCases: [stack],
+});
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts:28]: There are no assertions here -- can you document why you think deployment of the stack is sufficient to test the two custom resources?

File: packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
 
 class TestStack extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
-    // FIXME: currently this is only released as part of amplify-alpha, which cannot be tested here
-    // and the asset hash is different in alpha
+    // Deploying this integ test would create an Amplify app with branch name 'main'
+    // Re-deploying the stack again after updating the asset file would update the Amplify app.
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts:13]:
```suggestion
    // Deploying this integ test create san Amplify app with branch name 'main'
    // Re-deploying the stack again after updating the asset file updates the Amplify app.
```

Comment (kaizencc) [reply packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts:13]: Can you explain why this is relevant as well?

**Review by kaizencc — APPROVED** [review]
Conditionally approved - would like to see some additional documentation convincing me that this tests the 2 custom resources in amplify.

File: packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts (hunk 2 above)
Comment (SankyRed) [reply packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts]: Wanted to do some clean up on Cory's comments for the test doesn't work which was from a while ago. But, removed it.

File: packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts (hunk 1 above)
Comment (SankyRed) [reply packages/@aws-cdk/aws-amplify-alpha/test/integ.app-asset-deployment.ts:29]: Did add some comments explaining the working of custom resources as expected with stack deployment.

**Pull request merged**